README
Net-NSCA-Client 0.008
=====================

INSTALLATION
------------

To install this module, run the following commands:

    perl Build.PL
    ./Build
    ./Build test
    ./Build install

SCRIPTS
-------

This distribution comes with useful scripts in the `bin/` directory which
you may want to use.

SUPPORT AND DOCUMENTATION
-------------------------

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Net::NSCA::Client

You can also look for information at:

*   [RT, CPAN's request tracker](http://rt.cpan.org/NoAuth/Bugs.html?Dist=Net-NSCA-Client)

*   [AnnoCPAN, Annotated CPAN documentation](http://annocpan.org/dist/Net-NSCA-Client)

*   [CPAN Ratings](http://cpanratings.perl.org/d/Net-NSCA-Client)

*   [Search CPAN](http://search.cpan.org/dist/Net-NSCA-Client)

COPYRIGHT AND LICENCE
---------------------

Copyright 2009 Douglas Christopher Wilson.

This program is free software; you can redistribute it and/or modify it under
the terms of either:

*   the GNU General Public License as published by the Free Software Foundation;
    either version 1, or (at your option) any later version, or
*   the Artistic License version 2.0.


Changes
Revision history for Perl 5 distribution Net-NSCA-Client

0.009001 2011-05-05
 [BUG FIXES]
 - The test t/live_server.t had a test at the beginning to test for the
   exception of connecting to nothing. Since these error messages are not
   the same always, the test is changed to just test for any exception.

0.009 2011-05-02
 [ENHANCEMENTS]
 - A specific error message will occur if the remote server disconnects
   early.
 - Add a troubleshooting guide.
 - Lots of new tests; very close to 100% coverage.

 [BUG FIXES]
 - An attribute was missing the lazy attribute and Moose 2.00 caused this
   to make the tests fail. Added the lazy attribute.
 - When the initialization_vector_length is changed in ServerConfig the new
   value is honored.

 [OTHER]
 - Change all uses of Readonly to Const::Fast.
 - Change all uses of Test::Exception to Test::Fatal.

0.008 2010-08-30
 [BUG FIXES]
 - Actually sending a report was broken with a message about not being able
   to clone something.

0.007 2010-08-24
 [ENHANCEMENTS]
 - Allow communicating with NSCA servers that have been custom-compiled
   with different packet constants.
 - Check for data integrety when decoding packets.
 - Fill empty space in packets with random alpha numeric data as the
   standard NSCA client/server does.

 [BUG FIXES]
  - Remove test that failed when the data "didn't look random".

 [OTHER]
 - Change from Module::Install to Module::Build.

0.006 2009-11-03
 [BUG FIXES]
 - Fixed an issue where the client would try to parse the initial packet
   too early and throw an error about the UNIX timestamp not meeting a type
   constraint.

0.005 2009-11-01
 [BUG FIXES]
 - Finally fixed the test failures for t/send_nsca.t by requiring
   Pod::Usage 1.36+ for the script.

0.004 2009-10-31
 [BUG FIXES]
 - Modified t/send_nsca.t test so that the non-present required modules for
   the utility will not cause the entire module to fail.

0.003 2009-10-31
 [ENHANCEMENTS]
 - New bin/send_nsca utility. This is an emulation of the send_nsca utility
   that comes with the NSCA package.

 [OTHER]
 - No longer use Test::Most in tests, as it was unnecessary.
 - Use Data::Rand::Obscure instead of Crypt::Random for generation of
   random IV.

0.002 2009-10-08
 [ENHANCEMENTS]
 - Added a public constant $INITIALIZATION_VECTOR_LENGTH to
   Net::NSCA::Client::InitialPacket.
 - Added tests for many pieces of the objects and for packet reading and
   creation.
 - The initialization_vector attribute in Net::NSCA::Client::InitialPacket
   is now always exactly 128 bytes.

 [BUG FIXES]
 - All Moose classes are now immutable.
 - Creating a Net::NSCA::Client::InitialPacket without an IV would cause an
   error since the default ended up being a Math::Pari object.

 [OTHER]
 - Added SEE ALSO section in Net::NSCA::Client.
 - Fixed documentation in Net::NSCA::Client::DataPacket to specify all
   required attributes.

0.001 2009-10-01
 - First release.


Makefile.PL
use inc::Module::Install;

name     'Net-NSCA-Client';
all_from 'lib/Net/NSCA/Client.pm';
author   'Douglas Christopher Wilson <doug@somethingdoug.com>';
license  'perl';

# Resources
resources homepage   => 'http://github.com/dougwilson/perl5-net-nsca-client/';
resources repository => 'git://github.com/dougwilson/perl5-net-nsca-client.git';

# Modules that are required
requires
	'Convert::Binary::C'        => '0.74',
	'Data::Rand::Obscure'       => '0.020',
	'Data::Validate::Domain'    => '0.02',
	'Digest::CRC'               => 0,
	'English'                   => 0,
	'IO::Socket::INET'          => 0,
	'Moose'                     => '0.89',
	'MooseX::Clone'             => 0,
	'MooseX::StrictConstructor' => '0.08',
	'MooseX::Types'             => '0.08',
	'Readonly'                  => '1.03',
	'namespace::clean'          => '0.04';

# Modules that are required for tests
test_requires
	'Test::Class'     => '0.33',
	'Test::Command'   => '0.08',
	'Test::Exception' => '0.03',
	'Test::More'      => '0.94',
	'Test::Requires'  => '0.02';

# Modules that are required for author tests
author_requires
	'File::Spec'                      => 0,
	'Module::Install'                 => '0.91',
	'Module::Install::AuthorRequires' => '0.02',
	'Module::Install::AuthorTests'    => '0.002',
	'Pod::Coverage'                   => '0.18',
	'Test::CleanNamespaces'           => 0,
	'Test::ConsistentVersion'         => 0,
	'Test::DistManifest'              => 0,
	'Test::MinimumVersion'            => '0.009',
	'Test::Perl::Critic'              => '1.01',
	'Test::Pod'                       => '1.22',
	'Test::Pod::Content'              => 0,
	'Test::Pod::Coverage'             => '1.08';

# Enable tests to be in multi-level directories
tests_recursive;

# Author tests in xt/
recursive_author_tests('xt');

# Write the provided classes in the META
auto_provides;

WriteAll;
